src/ast.zig

```diff
@@ -1520,15 +1520,38 @@ pub fn nodeChildrenAlloc(allocator: std.mem.Allocator, tree: Ast, node: Ast.Node
         children: *std.ArrayListUnmanaged(Ast.Node.Index),
         fn callback(self: @This(), ast: Ast, child_node: Ast.Node.Index) error{OutOfMemory}!void {
             _ = ast;
-            if (child_node == 0) return;
+            if (child_node == .root) return;
```

Comment:
The old meaning of `0` isn't always the same as `.root`. Previously, `0` could mean either `.root` or a "null" node. These two possibilities are now differentiated between through the type system with `Ast.Node.OptionalIndex`. It is impossible that the child node is `.root` so this check is no longer necessary:

```suggestion
```